lib/elixir/lib/map.ex

```diff
@@ -77,7 +77,10 @@ defmodule Map do
   """
   @spec new(Enum.t) :: map
   def new(enumerable) do
-    Enum.reduce(enumerable, %{}, fn {k, v}, acc -> put(acc, k, v) end)
+    enumerable
+    |> Enum.reduce([], &[&1 | &2])
+    |> :lists.reverse
```

Comment:
Ah whoops I just edited this from the MapSet version, that would indeed make sense.